core/src/main/java/org/mule/runtime/core/internal/source/scheduler/DefaultSchedulerMessageSource.java

```diff
@@ -157,37 +162,30 @@ private void poll() {
     }
 
     if (execute) {
-      Message request = of(null);
-      pollWith(request);
+      doPoll();
     } else {
       LOGGER.info("Flow '{}' is already running and 'disallowConcurrentExecution' is set to 'true'. Execution skipped.",
                   flowConstruct.getRootContainerLocation().getGlobalName());
     }
   }
 
-  private void pollWith(final Message request) {
+  private void doPoll() {
     try {
-      just(request)
-          .map(message -> InternalEvent.builder(create(flowConstruct, getLocation())).message(request)
-              .build())
-          .doOnNext(event -> setCurrentEvent(event))
-          .doOnNext(event -> notificationHelper.fireNotification(this, event, getLocation(), MESSAGE_RECEIVED))
-          .cast(CoreEvent.class)
-          .transform(listener)
-          .doOnError(MessagingException.class,
-                     me -> ((BaseEventContext) me.getEvent().getContext()).error(me))
-          .doOnSuccess(result -> ((BaseEventContext) result.getContext()).success())
-          .doFinally(s -> {
-            synchronized (DefaultSchedulerMessageSource.this) {
-              executing = false;
-            }
-          })
-          .subscribe(requestUnbounded());
+      setCurrentEvent(InternalEvent.builder(EventContextFactory.create(flowConstruct, getLocation())).message(Message.of(null))
```

Comment:
review if this is needed or already done by the ModuleFlowProcessingPhase

Reply:
It's not done in ModuleFlowProcessingPhase. The current event is set in the interceptors applied to each processor. In my opinion is needed here.